encodings/fsst/src/compute/compare.rs

```diff
@@ -16,33 +17,71 @@ impl CompareFn<FSSTArray> for FSSTEncoding {
         rhs: &Array,
         operator: Operator,
     ) -> VortexResult<Option<Array>> {
-        match (rhs.as_constant(), operator) {
-            (Some(constant), _) if constant.is_null() => {
+        match rhs.as_constant() {
```

Comment:
Can this not go in `pub fn compare`?

Reply:
its nice to have all the FSST-const logic in one function